crazystocks/ARCHITECTURE.md

```diff
@@ -0,0 +1,105 @@
+# Goals
+
+The goal is to produce a small toolkit and library for evaluating stock market strategies based on predictive approaches.
+
+By predictive approaches, we mean machine learning on quantities related to the stock market. The quantities may include prices, volumes, expected gains/losses under some action.
+
+By stragegies, we mean simple rule based strategies which use the predictions of the machine learning models
+and may apply simple techniques as thresholding, stop-loss rules, time of day and so on.
+
+The same code should be able to run on:
+
+- the analyst's laptop using the DBG public dataset
+- the quantopian.com website
+
+# Stretch goals
+
+Ideally, we should be able to demonstrate consistent gains on the models/strategies across multiple stocks
+and also on quantopian.com. The strategies should be robust to slippage and costs.
+
+# Non-goals
+
+We don't aim to apply the most sophisticated modeling approaches (e.g. LSTM, multi-headed Attention Mechanism, very deep neural networks) at the intial stage. We use two basic models Linear Regression and Boosted Decision Trees together
+with informative features.
+
+Our focus is on robustness and usability of code. We want to enable ease of experimentation and ease of validation on quantopian.com. 
```

Comment:
I wouldn't build anything to a proprietary framework. This will drive away institutional clients who (rightfully so) are afraid quantopian is stealing their ideas (which they probably are).

Reply:
I am not binding, just will run strategies there. I'll make it more clear.